tests/churn/test_churn.py

```diff
@@ -181,27 +249,54 @@ def multi_profile_df(generate_data):
     })
 
     snippets = [user_0, user_1, user_2]
-    return generate_data(snippets)
+    return (
+        generate_main_summary_data(snippets)
+        .withColumn("is_new_profile", F.lit(False))
+    )
 
 
-def test_extract_subset(generate_data):
-    df = churn.extract_subset(generate_data(None), week_start_ds, 1, 0, False)
+def test_extract_main_summary(spark, generate_main_summary_data):
+    df = churn.extract(
+        generate_main_summary_data(None),
+        spark.createDataFrame([], new_profile_schema),
+        week_start_ds, 1, 0, False
+    )
     assert df.count() == 1
 
 
-def test_ignored_submissions_outside_of_period(generate_data):
+def test_extract_new_profile(spark, generate_new_profile_data):
```

Comment:
Please add a test for `extract` that uses both `main_summary` and `new_profile`. Ideally covering both the case where `new_profile` contains the same `client_id`s as well as different `client_id`s

Reply:
Good call, I realized that the `is_new_profile` column can't be used to distinguish the source of the clients because a client can be in both categories. This shouldn't be a problem with HLL, but it can't be used here without also including a "all" value.